src/shared-components/alert/index.tsx

```diff
@@ -59,30 +59,33 @@ export const Alert: Alert = (alertProps) => {
     content,
     ctaContent = null,
     duration = 3,
-    onExit = () => undefined,
+    onExit = undefined,
     truncateText = false,
     type = 'default',
     ...rest
   } = alertProps;
   const theme = useTheme();
   const [exiting, setExiting] = useState(false);
   const [exited, setExited] = useState(false);
+  const timer = React.useRef<number | undefined>(undefined);
 
   const { contentText } = useTruncateText(truncateText);
 
-  let timer: number | undefined;
-
-  const alertExitHandler = () => {
+  const alertExitHandler = React.useCallback(() => {
     setExiting(true);
-    window.clearTimeout(timer);
 
-    const { onExit: _onExit, ...otherProps } = alertProps;
+    if (isDefined(timer.current)) {
+      window.clearTimeout(timer.current);
```

Comment:
There's no bugs from calling `window.clearTimeout(undefined)` but this just makes the `number | undefined` relationship more explicit. Using a ref also simplifies our dependency array for `useCallback` since we would otherwise have to add it.